harpoon/internal/testing/panic.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package testing
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+)
+
+const godogFailMessage = "FailNow or SkipNow called"
+
+type stringer interface {
+	String() string
+}
+
+func WrapWithPanicHandler[U any](needsLog bool, exitBehavior ExitBehavior, fn U) U {
```

Comment:
This would definitely benefit from a few comments 😅

Reply:
I'll add some before merging, I probably should have initially since it's a pretty gnarly function.